js/signin.js

```diff
@@ -0,0 +1,64 @@
+const emailErrorMessage = document.querySelector(".email-error-message");
+const emailInput = document.querySelector(".email-input");
+
+emailInput.addEventListener("blur", function () {
+  const EMAIL_REGEX = /[a-z0-9]+@[a-z]+\.[a-z]{2,3}/
+
+  if (emailInput.value === "") {
+    emailErrorMessage.textContent = "이메일을 입력해 주세요.";
+    emailInput.classList.add("error-border");
+    return;
+  }
+
+  if (emailInput.value.match(EMAIL_REGEX) === null) {
+    emailErrorMessage.textContent = "올바른 이메일 주소가 아닙니다.";
+    emailInput.classList.add("error-border");
+    return;
+  }
+
+  emailErrorMessage.textContent = "";
+  emailInput.classList.remove("error-border");
+});
+
+const passwordErrorMessage = document.querySelector(".password-error-message");
+const passwordInput = document.querySelector(".password-input");
+
+passwordInput.addEventListener("blur", function () {
+  if (passwordInput.value === "") {
+    passwordErrorMessage.textContent = "비밀번호를 입력해 주세요.";
+    passwordInput.classList.add("error-border");
+    return;
+  }
+
+  passwordErrorMessage.textContent = "";
+  passwordInput.classList.remove("error-border");
+});
+
+
+const eyeIconChange = document.querySelector(".eye-icon-change");
+
+eyeIconChange.addEventListener("click", function () {
+  if (passwordInput.type === "password"){
+    eyeIconChange.src = "./image/eye-on.png";
+    passwordInput.type = "text";
+    return;
+  }
+  eyeIconChange.src = "./image/eye-off.png";
+    passwordInput.type = "password";
+});
+
+const loginForm = document.querySelector(".login-form");
+
+loginForm.addEventListener("submit", function (event) {
+  event.preventDefault();
+  if (emailInput.value !== "test@codeit.com" || passwordInput.value !== "codeit101") {
```

Comment:
"test@codeit.com", "codeit101"과 같은 문자열도 상수화를 고민해볼만한 포인트일 것 같아요!

Reply:
조건식을 이렇게 변수로 추출해볼수도 있는데요~
조건식을 깔끔하게 작성하기 위한 하나의 기법이라고 생각해주시면 좋을 것 같아요!
`리팩토링2판`같은 책에서도 확인하실 수 있습니다!

```suggestion
  event.preventDefault();
  
  const isMaster = emailInput.value === "test@codeit.com" && passwordInput.value === "codeit101";
  
  if (isMaster === false) {
```